### PR TITLE
#2580 Implement better detection of "multipart/encrypted" email with PGP encrypted message

### LINKED ISSUE #2580: Implement better detection of "multipart/encrypted" email with PGP encrypted message
Description: The FlowCrypt iOS didn't recognize "multipart/encrypted" email with PGP encrypted message. Emails at https://mail.google.com/mail/u/flowcrypt.compatibility@gmail.com/#inbox/FMfcgzGxTFXMFnDNbgknpqvXKtSmMgSK isn't recognized by the FlowCrypt app and shows not encrypted + no signed blank email thread.

Reference: 
- https://github.com/FlowCrypt/flowcrypt-browser/pull/5727
- https://github.com/FlowCrypt/flowcrypt-android/blob/e58d040f8eade675ca8c6bd18cb608d46bcdb44b/FlowCrypt/src/main/java/com/flowcrypt/email/extensions/jakarta/mail/PartExt.kt#L82

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR implemented better detection of "multipart/encrypted" email with PGP encrypted message

close #2580 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCrypt/Functionality/Mail Provider/Message Provider/Gmail+MessageExtension.swift [hunk 1]
```diff
@@ -148,12 +148,12 @@ private extension GTLRGmail_Message {
 
     func parseAttachments() -> [MessageAttachment] {
         attachmentParts.compactMap {
```
Comment (sosnovsky) [inline FlowCrypt/Functionality/Mail Provider/Message Provider/Gmail+MessageExtension.swift:150]: probably separate `attachmentParts` property is not needed anymore, as it just takes original `payload.parts` array, and can be replaced directly with `payload?.parts?.compactMap { ... } ?? []`

File: appium/tests/specs/mock/inbox/CheckCanaryMail.spec.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { MockApi } from 'api-mocks/mock';
+import { MockApiConfig } from 'api-mocks/mock-config';
+import { EmailScreen, MailFolderScreen, SetupKeyScreen, SplashScreen } from '../../../screenobjects/all-screens';
+
+describe('INBOX: ', () => {
+  it('user is able to view correct signature badge for all cases', async () => {
```
Comment (sosnovsky) [inline appium/tests/specs/mock/inbox/CheckCanaryMail.spec.ts:6]: incorrect test title :)

**Review by sosnovsky — COMMENTED** [review]
Works well 👍

File: appium/tests/specs/mock/inbox/CheckCanaryMail.spec.ts (hunk 2 above)
Comment (ioanmo226) [reply appium/tests/specs/mock/inbox/CheckCanaryMail.spec.ts:6]: Oops.

**Review by sosnovsky — APPROVED** [review]
👍

**Pull request merged**